src/validation.cpp

```diff
@@ -3113,7 +3114,7 @@ bool Chainstate::ConnectTip(
     LogDebug(BCLog::BENCH, "  - Load block from disk: %.2fms\n",
              Ticks<MillisecondsDouble>(time_2 - time_1));
     {
-        CCoinsViewCache view(&CoinsTip());
+        auto& view{*m_coins_views->m_connect_block_view};
```

Comment:
14f1b7913861d95e3ffeda658d45b0b88e7019e7:

I love this separation, we could extract it to a new PR since it kinda' makes sense on its own - carving out a small portion of this PR.

Could we add a `Start` here as well, which could assert that this was in a reset state at the beginning?
Besides the symmetry with `Reset`, it could help with using this same view for the other `/*will_reuse_cache=*/false`. The assertion would assure that the two cannot accidentally run at the same time (proving that we can reuse the same instance)

Reply:
Hmm doing `SetBestBlock(base->GetBestBlock());` seems like a behavior change which I don't want to do here. `Reset` is basically returning the state of `CCoinsViewCache` to a fresh copy, so it is identical behavior-wise to what we had before.

If we don't call `SetBestBlock`, then I don't see a benefit to including a `Start` method.